### PR TITLE
Moves the rest of farm animals over to basic animals except for goats, special AI for deer

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Moves farm animals over to basic mobs

Deer are easily spooked and they'll run away from any non deer.

## Why It's Good For The Game
The only thing I really have to justify is giving deer special AI:

Deer are used in a single spot in the entire game and they have zero special traits

## Changelog

:cl:
refactor: Chicks, Chickens, and Deer are now basic mobs codewise.
expansion: This means they'll behave a little different!
/:cl:

### CODE REVIEW

File: code/modules/mob/living/basic/farm_animals/chickens.dm [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+/mob/living/basic/chick/proc/hologram_check()
+	if(!(flags_1 & HOLOGRAM_1))
+		AddComponent(/datum/component/growing_up, /mob/living/basic/chicken, 100)
+
+/datum/ai_controller/basic_controller/chick
+
+	ai_traits = STOP_MOVING_WHEN_PULLED
+	ai_movement = /datum/ai_movement/basic_avoidance
+
+	//IMPLEMENT IDLE BEHAVIOR HERE
```
Comment (tralezab) [inline code/modules/mob/living/basic/farm_animals/chickens.dm:50]: requires https://github.com/tgstation/tgstation/pull/61455,

will allow for a generic idle datum

File: code/modules/mob/living/basic/farm_animals/chickens.dm [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/random_speech/chick,
+	)
+
+//REMOVE THIS
```
Comment (tralezab) [inline code/modules/mob/living/basic/farm_animals/chickens.dm:56]: remove this when doing above

File: code/modules/mob/living/basic/farm_animals/chickens.dm [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+/mob/living/basic/chicken/proc/egg_laid(obj/item/egg)
+	if(chicken_count <= MAX_CHICKENS && process_eggs && prob(25))
+		START_PROCESSING(SSobj, egg)
+
+/datum/ai_controller/basic_controller/chicken
+
+	ai_traits = STOP_MOVING_WHEN_PULLED
+	ai_movement = /datum/ai_movement/basic_avoidance
+
+	//IMPLEMENT IDLE BEHAVIOR HERE
```
Comment (tralezab) [inline code/modules/mob/living/basic/farm_animals/chickens.dm:129]: requires https://github.com/tgstation/tgstation/pull/61455,

will allow for a generic idle datum

File: code/modules/mob/living/basic/farm_animals/chickens.dm [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/random_speech/chicken,
+	)
+
+//REMOVE THIS
+/datum/ai_controller/basic_controller/chicken/PerformIdleBehavior(delta_time)
```
Comment (tralezab) [inline code/modules/mob/living/basic/farm_animals/chickens.dm:136]: remove this when doing above

File: code/modules/mob/living/basic/farm_animals/deer.dm [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+/datum/ai_controller/basic_controller/deer
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic,
+	)
+
+	ai_traits = STOP_MOVING_WHEN_PULLED
+	ai_movement = /datum/ai_movement/basic_avoidance
+
+	//IMPLEMENT IDLE BEHAVIOR HERE
```
Comment (tralezab) [inline code/modules/mob/living/basic/farm_animals/deer.dm:38]: requires https://github.com/tgstation/tgstation/pull/61455,

will allow for a generic idle datum

File: code/modules/mob/living/basic/farm_animals/deer.dm [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/random_speech/deer,
+		/datum/ai_planning_subtree/simple_find_target/close,
+		/datum/ai_planning_subtree/run_away,
+	)
+
+//REMOVE THIS
+/datum/ai_controller/basic_controller/deer/PerformIdleBehavior(delta_time)
```
Comment (tralezab) [inline code/modules/mob/living/basic/farm_animals/deer.dm:47]: remove this when doing above

**Review by tralezab — COMMENTED** [review]
good work bazelart but i have some notes

**Comment by JohnFulpWillard** [comment]
I dont know if this is a good time to mention it, but vim mech is hardcoded to only allow simple_mobs

**Comment by tralezab** [comment]
good enough time!

File: code/datums/ai/basic_mobs/basic_ai_behaviors/run_away.dm [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+	controller.pawn.visible_message(span_warning("[controller.pawn] [alarmed_message]"))
+
+/datum/ai_behavior/run_away/perform(delta_time, datum/ai_controller/controller, target_key, alarmed_message)
+	var/atom/target = controller.blackboard[target_key]
+	if(!target) //?!
+		finish_action(controller, TRUE)
+	if(get_dist(controller.pawn, target) >= 15)
+		finish_action(controller, TRUE)
+	else
+		controller.current_movement_target = get_step_away(controller.pawn, target)
```
Comment (AMonkeyThatCodes) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/run_away.dm:17]: I'm not a huge fan of fleeing being it's own behavior, but its okay for now. I'll probably make this into something generic down the line as ranged mobs also need this.

File: code/datums/ai/basic_mobs/basic_subtrees/simple_find_target.dm [hunk 8]
```diff
@@ -1,8 +1,12 @@
...
 /datum/ai_planning_subtree/simple_find_target/SelectBehaviors(datum/ai_controller/controller, delta_time)
 	. = ..()
 	var/atom/target = controller.blackboard[BB_BASIC_MOB_CURRENT_TARGET]
 	if(target && !QDELETED(target))
 		return
-	controller.queue_behavior(/datum/ai_behavior/find_potential_targets, BB_BASIC_MOB_CURRENT_TARGET, BB_TARGETTING_DATUM, BB_BASIC_MOB_CURRENT_TARGET_HIDING_LOCATION)
+	controller.queue_behavior(search_behavior, BB_BASIC_MOB_CURRENT_TARGET, BB_TARGETTING_DATUM, BB_BASIC_MOB_CURRENT_TARGET_HIDING_LOCATION)
+
+/datum/ai_planning_subtree/simple_find_target/close
+	search_behavior = /datum/ai_behavior/find_potential_targets/too_close
```
Comment (AMonkeyThatCodes) [inline code/datums/ai/basic_mobs/basic_subtrees/simple_find_target.dm:12]: Make vision range into a blackboard key instead and just make people define it on the mob they're making

File: code/game/objects/items/food/egg.dm [hunk 9]
```diff
@@ -19,7 +19,21 @@
 	microwaved_type = /obj/item/food/boiledegg
 	foodtypes = MEAT
 	w_class = WEIGHT_CLASS_TINY
-	var/static/chick_count = 0 //I copied this from the chicken_count (note the "en" in there) variable from chicken code.
+	///I copied this from the chicken_count (note the "en" in there) variable from chicken code.
+	var/static/chick_count = 0
+	///How much this
```
Comment (AMonkeyThatCodes) [inline code/game/objects/items/food/egg.dm:24]: hmmmm

**Pull request closed without merging**